source/write/replace.txt

```diff
@@ -0,0 +1,282 @@
+       - :dbcommand:`currentOp` output, in the :data:`currentOp.command
+         <currentOp.command>` ``comment`` field
+
+       | For more information, see the :manual:`insert Command
+         Fields guide </reference/command/insert/#command-fields>` in the {+mdb-server+} manual.
+
+   * - ``hint``
+     - | Specifies the index to scan for documents that match the query filter.
+         For more information, see the
+         :manual:`Update Statement fields guide </reference/command/update/#std-label-update-statement-documents>`
+         in the {+mdb-server+} manual. 
+
+   * - ``let``
+     - | Specifies a document containing variables and their values to be used in the ``replace_one()`` method. 
+         This allows you to improve code readability by separating the variables from the operation text. 
+         Values must be constant or closed expressions that don't reference document fields. 
+         For more information, see the 
+         :manual:`update Command Fields guide </reference/command/update/#std-label-replace-let-syntax>`
+         in the {+mdb-server+} manual.
+
+   * - ``upsert``
+     - | Specifies whether the replace operation performs an upsert operation if no 
+         documents match the query filter.
+       | Defaults to ``false``.
+
+   * - ``write_concern``
+     - | Sets the write concern for the operation.
+         For more information, see :manual:`Write Concern </reference/write-concern/>`
+         in the {+mdb-server+} manual.
+
+Example: ``hint``
+^^^^^^^^^^^^^^^^^
+
+The following example uses the :ref:`create_index() <cpp-indexes>` method to create an ascending single-field index 
+on the ``name`` field. It then instantiates a ``mongocxx::options::replace`` object called ``opts`` and sets its ``hints`` field value to the name 
+of the new index, ``name_1``. When ``opts`` is passed in as an optional argument, the ``replace_one()`` method searches the ``name_1`` index for a
+``name`` field value of ``Nobu`` and replaces the associated document with a new document that has a ``name`` field value of ``La Bernadin``.  
+
+.. literalinclude:: /includes/write/replace.cpp
+   :start-after: start-replace-options-hint
+   :end-before: end-replace-options-hint
+   :language: cpp
+   :dedent:
+
+To learn more about indexes, see :ref:`Optimize Queries with Indexes <cpp-indexes>`.
+
+Example: ``upsert`` 
+^^^^^^^^^^^^^^^^^^^
+
+The following example instantiates a ``mongocxx::options::replace`` object called ``opts``
+and sets its ``upsert`` field value to ``true``.  When ``opts`` is passed in as an optional argument, the ``replace_one()`` method
+searches the ``sample_restaurants.restaurants`` collection for a document with a ``name`` field value of ``In-N-Out Burger``. 
+Because this sample collection does not contain a document matching this query filter,
+the ``replace_one()`` method upserts a new document with a ``name`` field value of ``Shake Shack`` into the collection. 
```

Comment:
S: similar to my suggestion above, I'd shorten this description to something resembling:
```suggestion
The following example passes a ``mongocxx::options::replace`` object to the ``replace_one()`` method and sets its ``upsert`` field value to ``true``.  Because no documents match the query filter, this instructs the operation to insert a new document with a ``name`` field value of ``"Shake Shack"`` into the collection:
```